core/dbt/exceptions.py

```diff
@@ -1961,6 +1961,21 @@ def get_message(self) -> str:
         return msg
 
 
+class AmbiguousResourceNameRefError(CompilationError):
+    def __init__(self, duped_name, unique_ids, node=None):
+        self.duped_name = duped_name
+        self.unique_ids = unique_ids
+        self.packages = [unique_id.split(".")[1] for unique_id in unique_ids]
+        super().__init__(msg=self.get_message(), node=node)
+
+    def get_message(self) -> str:
+        msg = (
+            f'dbt found multiple potential nodes when referencing "{self.duped_name}" - {self.unique_ids}.'
+            f"\nTo fix this, specify which package to use as the first argument to 'ref' (one of: {self.packages})."
```

Comment:
This is great! Teensy nits around the formatting of our lists:
```
Compilation Error in model model_downstream (models/model_downstream.sql)
  dbt found multiple potential nodes when referencing 'model': 'model.project_b.model', 'model.project_a.model'
  To fix this, use two-argument 'ref' with the package name ('project_b' or 'project_a')
```

I think a join on `"or"` is fine even if there's >2:
```
('project_a' or 'project_b' or 'project_c' ...)
```

Reply:
Thank you, updated!